File: src/main/kotlin/dev/jefy/connectpro/user/applicaion/dtos/ActivateAccountRequest.java
package dev.jefy.connectpro.user.applicaion.dtos;


import org.jspecify.annotations.NullMarked;

import java.util.UUID;

import jakarta.validation.constraints.NotNull;

/**
 * @author Jôph Yamba
 */

@NullMarked
public record ActivateAccountRequest(@NotNull UUID tokenId, @NotNull int code) {}

File: src/main/kotlin/dev/jefy/connectpro/user/application/dtos/ActivateAccountRequest.kt
package dev.jefy.connectpro.user.application.dtos

import jakarta.validation.constraints.NotBlank
import org.hibernate.validator.constraints.Length
import java.util.*

data class ActivateAccountRequest(
    val tokenId: UUID, 
    @field:NotBlank(message = "otp code must not be blank")
    @field:Length(min = 4, max = 4)
    val code: String)